[flang][OpenMP] Privatize locally destroyed values in `do concurent` (#112)
ROCm/llvm-project: flang/docs/DoConcurrentConversionToOpenMP.md

By default, variables that are used inside a `do concurernt` loop nest are
either treated as `shared` in case of mapping to `host`, or mapped into the
`target` region using a `map` clause in case of mapping to `device`. The only
exceptions to this are:

>   1. the loop's iteration variable(s) (IV) of **perfect** loop nest. In that

Reviewer: ```suggestion
  1. the loop's iteration variable(s) (IV) of **perfect** loop nests. In that
```

     case, for each IV, we allocate a local copy as shown the by the mapping
     examples above.

>   1. any values that are allocation outside the loop nest and destroyed inside

Reviewer: ```suggestion
  1. any values that are from allocations outside the loop nest and destroyed inside
```

>      explicitly inside of it. In such cases, a local privatized value is

Reviewer 2: ```suggestion
     exclusively inside of it. In such cases, a local privatized value is
```
